### PR TITLE
[WIP] Babel 6

### PR DESCRIPTION
**Pull request by Turbo87**
/cc @stefanpenner @rwjblue

### CODE REVIEW

**Comment by rwjblue** [comment]
Aside: Can we move `lib/utils/babel-enifed-module-formatter.js` into a stand alone babel plugin (ala `babel-plugin-filter-imports` and `babel-plugin-feature-flags`)?

File: lib/utils/transpile-es6.js [hunk 1]
```diff
@@ -40,12 +40,36 @@ module.exports = function(tree, description, _options) {
     options.plugins = options.plugins.slice();
```
Comment (rwjblue) [inline lib/utils/transpile-es6.js]: Is this stil correct?  It looks like these should either be using `options.firstPass` or `options.secondPass`, or is there also still a `options.plugins`?

**Comment by Turbo87** [comment]
> Aside: Can we move lib/utils/babel-enifed-module-formatter.js into a stand alone babel plugin (ala babel-plugin-filter-imports and babel-plugin-feature-flags)?

that would probably be a good idea. but they should still be integration tested in combination with the other plugins. as you can see from the code some of them don't play nicely with each other. I thought the AMD transform could be run in the same pass as the `enifed` formatter, but I couldn't make it work so far.

File: lib/utils/transpile-es6.js (hunk 1 above)
Comment (Turbo87) [inline lib/utils/transpile-es6.js]: right now there is also `options.plugins` which is part of the `WIP` in the title 😉 

I think that `options.plugins` should probably be appended to the first pass plugins

File: tests/expected/concat-tests/ember-tests.js [hunk 2]
```diff
@@ -1,141 +1,163 @@
...
-enifed("ember-metal/streams/simple", ["exports"], function (exports) {
-  "use strict";
-});
+//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbImVtYmVyLW1ldGFsL2JpbmRpbmcuanNoaW50LmpzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiI7OztBQUFBLFNBQU8sc0JBQVA7QUFDQSxPQUFLLDJDQUFMLEVBQWtELFlBQVc7QUFDM0QsT0FBRyxJQUFILEVBQVMsNENBQVQ7QUFDRCxHQUZEIiwiZmlsZSI6ImVtYmVyLW1ldGFsL2JpbmRpbmcuanNoaW50LmpzIiwic291cmNlc0NvbnRlbnQiOlsibW9kdWxlKCdKU0hpbnQgLSBlbWJlci1tZXRhbCcpO1xudGVzdCgnZW1iZXItbWV0YWwvYmluZGluZy5qcyBzaG91bGQgcGFzcyBqc2hpbnQnLCBmdW5jdGlvbigpIHsgXG4gIG9rKHRydWUsICdlbWJlci1tZXRhbC9iaW5kaW5nLmpzIHNob3VsZCBwYXNzIGpzaGludC4nKTsgXG59KTtcbiJdfQ==
+enifed("ember-metal/streams/simple", [], function () {});
 //# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IiIsImZpbGUiOiJlbWJlci1tZXRhbC9zdHJlYW1zL3NpbXBsZS5qcyIsInNvdXJjZXNDb250ZW50IjpbXX0=
-enifed('ember-metal/streams/simple.jshint', ['exports'], function (exports) {
+enifed('ember-metal/streams/simple.jshint', ['module'], function (module) {
```
Comment (rwjblue) [inline tests/expected/concat-tests/ember-tests.js]: We discussed this in slack, but just leaving a note here also: this is definitely an issue we have to address.  The `module` usage below is assuming a global module and shouldn't have been automatically wrapped with a non-existant import.

**Comment by rwjblue** [comment]
Thank you for working on this @Turbo87! Ember is now on babel@6 (as of https://github.com/emberjs/ember.js/pull/15037).

**Pull request closed without merging**